arch/arc/core/fast_irq.S

```diff
@@ -22,6 +22,36 @@
 GTEXT(_firq_enter)
 GTEXT(_firq_exit)
 
+#if defined(CONFIG_ARC_FIRQ_STACK)
+GTEXT(z_arc_firq_stack_set)
+SECTION_FUNC(TEXT, z_arc_firq_stack_set)
+	mov_s r0, _firq_interrupt_stack
+#ifdef CONFIG_SMP
+	_get_cpu_id r1
+	mpy r1, r1, CONFIG_ARC_FIRQ_STACK_SIZE
```

Comment:
This assumes there is a hw multiplier configured - probably true for most ARC instances, but doesn't have to be...

Reply:
It's a C arrary operatioon. It's a fast way to do this using mpy instruction. Without it, code will become very complex.  I think when firq is configured and enabled, mpy should be there.

Reply:
@vonhust it that case it worth checking explicitly if `__ARC_MPY__` is defined by GCC and report user an error due to building for unsupported configuration. Otherwise user might be surprised be the compiler error saying something about unknown opcode or something.

Reply:
@abrodkin  mpy is used when SMP and FIRQ  and FIRQ_STACK are all enabled, for this case, it doesn't make sense that mpy is not there. So  do not want to add __ARC_MPY__ for it to make codes complex.

Reply:
@vonhust As I just learned there's a bigger problem here! Our GCC only checks supported opcodes when it compiles C! But when only assembler & linker are involved user will only know this instruction is not supported on execution when exception happens. Thus I strongly suggest to add a check for mentioned `__ARC_MPY__ ` definition.

Reply:
@abrodkin OK, I turn z_arc_firq_stack_set into a C function with inline assembly,  don't use mpy instruction explictly.